demo/sample_azure_iot_embedded_sdk_pkcs11/common/demo_key_provisioning.c

```diff
@@ -0,0 +1,574 @@
+        }
+    }
+
+    mbedtls_pk_free( &xMbedPkContext );
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/* Delete well-known crypto objects from storage. */
+static CK_RV xDestroyDefaultCryptoObjects( CK_SESSION_HANDLE xSession )
+{
+    CK_RV xResult;
+    CK_BYTE * pxPkcsLabels[] =
+    {
+        ( CK_BYTE * ) pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS,
+        ( CK_BYTE * ) pkcs11configLABEL_CODE_VERIFICATION_KEY,
+        ( CK_BYTE * ) pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+        ( CK_BYTE * ) pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS
+    };
+    CK_OBJECT_CLASS xClass[] =
+    {
+        CKO_CERTIFICATE,
+        CKO_PUBLIC_KEY,
+        CKO_PRIVATE_KEY,
+        CKO_PUBLIC_KEY
+    };
+
+    xResult = xDestroyProvidedObjects( xSession,
+                                       pxPkcsLabels,
+                                       xClass,
+                                       sizeof( xClass ) / sizeof( CK_OBJECT_CLASS ) );
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+static uint32_t prvImportClientCred( CK_SESSION_HANDLE xSession )
+{
+    CK_RV xResult;
+    CK_FUNCTION_LIST_PTR pxFunctionList;
+    CK_OBJECT_HANDLE xObject = 0;
+    CK_BBOOL xImportedPrivateKey = CK_FALSE;
+    CK_BBOOL xKeyPairGenerationMode = CK_FALSE;
+
+#if ( pkcs11configIMPORT_PRIVATE_KEYS_SUPPORTED == 1 )
+    xResult = xDestroyDefaultCryptoObjects( xSession );
+
+    if( xResult != CKR_OK )
+    {
+        configPRINTF( ( "Warning: could not clean-up old crypto objects. %d \r\n", xResult ) );
+    }
+#endif
+
+    xResult = xProvisionCertificate( xSession,
+                                     democonfigCLIENT_CERTIFICATE_PEM,
+                                     sizeof( democonfigCLIENT_CERTIFICATE_PEM ),
+                                     ( uint8_t * ) pkcs11configLABEL_DEVICE_CERTIFICATE_FOR_TLS,
+                                     &xObject );
+
+    if( ( xResult != CKR_OK ) || ( xObject == CK_INVALID_HANDLE ) )
+    {
+        configPRINTF( ( "ERROR: Failed to provision device certificate. %d \r\n", xResult ) );
+        return xResult;
+    }
+
+#if ( pkcs11configIMPORT_PRIVATE_KEYS_SUPPORTED == 1 )
+    xResult = xProvisionPrivateKey( xSession,
+                                    democonfigCLIENT_PRIVATE_KEY_PEM,
+                                    sizeof( democonfigCLIENT_PRIVATE_KEY_PEM ),
+                                    ( uint8_t * ) pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+                                    &xObject );
+    if( ( xResult != CKR_OK ) || ( xObject == CK_INVALID_HANDLE ) )
+    {
+        configPRINTF( ( "ERROR: Failed to provision device private key with status %d.\r\n", xResult ) );
+        return xResult;
+    }
+#endif
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+uint32_t DemoKeyProvision()
+{
+    uint32_t xReturn = 0;
+    CK_RV xResult;
```

Comment:
nit: can this be named something like xPKCSResult. Just because at quick glance xReturn and xResult look verrrrry similar